### PR TITLE
Tpwdcee 6347 feature

### PR DESCRIPTION
**Pull request by tomazpu**

### CODE REVIEW

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Gateway/Response/ResponseHandler.php  1
- Controller/Frontend/Callback.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/magento2-ee/pullRequest?prid=5278355&bid=17283801)

File: Controller/Frontend/Callback.php [hunk 1]
```diff
@@ -162,14 +173,12 @@ private function isCreditCardThreeD()
     }
 
     /**
-     * @param string $redirectUrl
+     * @param array $data
      * @return Json
      * @since 3.1.2
      */
-    private function createRedirectResult($redirectUrl)
+    private function createRedirectResult($data)
```
Comment (edra-bumci) [inline Controller/Frontend/Callback.php]: @tomazpu maybe the data param should be named in a more descriptive way(is this formData)?

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Gateway/Response/ResponseHandler.php  1
- Controller/Frontend/Callback.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/magento2-ee/pullRequest?prid=5278355&bid=17283801)

File: Controller/Frontend/Redirect.php [hunk 2]
```diff
@@ -66,68 +60,87 @@ class Redirect extends Action implements CsrfAwareActionInterface
...
+    }
+
+    /**
+     * @param array $params
+     * @return bool
+     * @since 3.1.2
+     */
+    private function isCreditCardNonThreeDPayment($params)
+    {
+        if ($this->paymentMethodName === 'creditcard' && isset($params['data'])) {
```
Comment (edra-bumci) [inline Controller/Frontend/Redirect.php]: Just a question: Dont we have a place in our plugin for storing all payment method names as constants?

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Controller/Frontend/Callback.php  1
- Gateway/Response/ResponseHandler.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/magento2-ee/pullRequest?prid=5278355&bid=17283801)

**Pull request merged**